### PR TITLE
Feature kuiic

### PR DESCRIPTION
**Pull request by nxf58843**
Adding support for two K32L2B3 based boards, the FRDM-K32L2B3 and the KUIIC board.

This required an update to TinyUSB to get support for the K32L2B and I had to make some edits to other ports to fix the build. One notable edit is that I removed net_device.c from the builds since this file has been replaced by two different files and it is not clear if it is required or which of the new files to use.  The builds are working without this file included.

### CODE REVIEW

File: ports/k32l2/boards/frdm-k32l2b/board.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+// Number of neopixels
+#define NEOPIXEL_NUMBER       0
+
+//--------------------------------------------------------------------+
+// USB UF2
+//--------------------------------------------------------------------+
+
+// 0x1FC9, 0x0154 is reserved for TinyUF2 usage
+#define USB_VID           0x1FC9
+#define USB_PID           0x0154
```
Comment (hathach) [inline ports/k32l2/boards/frdm-k32l2b/board.h:67]: does NXP explicitly allocate this PID for TinyUF2 usage ?

**Review by hathach — COMMENTED** [review]
thank you very much for the PR, I couldn't find detail where NXP allocate the 0x0154 for TinyUF2 usage and it would cause pid conflict later in the future. Would you mind clarifying it a bit more.

File: ports/k32l2/boards/frdm-k32l2b/board.h (hunk 1 above)
Comment (nxf58843) [reply ports/k32l2/boards/frdm-k32l2b/board.h:67]: Yes, I requested a dedicated PID for TinyUF2 usage.
The plan is to use it for TinyUF2 on any NXP based board.

Comment (hathach) [reply ports/k32l2/boards/frdm-k32l2b/board.h:67]: thanks for your confirmation. Preferably each product/board will have its own PID, though sharing the PID for NXP MCU sound reasonable as well. Did NXP approve your request yet ?

**Comment by nxf58843** [comment]
Yes, that is the numbers they signed for this.
I can dm the message if that helps.
Thanks,
Greg

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Ha Thach ***@***.***>
Sent: Monday, December 20, 2021 8:44 PM
To: adafruit/tinyuf2
Cc: Greg Steiert; Author
Subject: [EXT] Re: [adafruit/tinyuf2] Feature kuiic (PR #168)


Caution: EXT Email

@hathach commented on this pull request.

________________________________

In ports/k32l2/boards/frdm-k32l2b/board.h<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fadafruit%2Ftinyuf2%2Fpull%2F168%23discussion_r772824241&data=04%7C01%7Cgreg.steiert%40nxp.com%7C768eb9195c294f4d2b2308d9c43c987a%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637756586818173285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=tMePQGvXHQGMylkX7miAHG3KDMs%2BQqBbrz%2B3%2BQQh4Co%3D&reserved=0>:

> +#define USB_VID           0x1FC9
+#define USB_PID           0x0154


thanks for your confirmation. Preferably each product/board will have its own PID, though sharing the PID for NXP MCU sound reasonable as well. Did NXP approve your request yet ?

—
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fadafruit%2Ftinyuf2%2Fpull%2F168%23discussion_r772824241&data=04%7C01%7Cgreg.steiert%40nxp.com%7C768eb9195c294f4d2b2308d9c43c987a%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637756586818173285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=tMePQGvXHQGMylkX7miAHG3KDMs%2BQqBbrz%2B3%2BQQh4Co%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOE6ZI2C2HQNUSBYL4TVSHTUSAA3LANCNFSM5KEZONAA&data=04%7C01%7Cgreg.steiert%40nxp.com%7C768eb9195c294f4d2b2308d9c43c987a%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637756586818173285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=BQ5zHLp%2FL2geT6W97Zvl6MOiayc8z9EVSJT6aBnsAbY%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cgreg.steiert%40nxp.com%7C768eb9195c294f4d2b2308d9c43c987a%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637756586818173285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=cWXpaGkFV8PrLKd%2F4CVJdFOuH8EgVDXBNGZVROIjBiA%3D&reserved=0> or Android<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cgreg.steiert%40nxp.com%7C768eb9195c294f4d2b2308d9c43c987a%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637756586818173285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=lE8mciqEXmUuDk6DSXVbXujD2ZTjWH17TGxAnyLCgCA%3D&reserved=0>.

File: ports/k32l2/boards/frdm-k32l2b/board.h (hunk 1 above)
Comment (hathach) [reply ports/k32l2/boards/frdm-k32l2b/board.h:67]: > Yes, that is the numbers they signed for this.
> I can dm the message if that helps.
> Thanks,
> Greg

thanks, this is good enough.

**Review by hathach — APPROVED** [review]
I have resolved the conflict with latest main, and push update. PR look great, thank you for you update. Will merge when ci passed.

**Pull request merged**